doc/source/whatsnew/v1.4.4.rst

```diff
@@ -19,6 +19,7 @@ Fixed regressions
 - Fixed regression in :meth:`DataFrame.loc` not updating the cache correctly after values were set (:issue:`47867`)
 - Fixed regression in :meth:`DataFrame.loc` not aligning index in some cases when setting a :class:`DataFrame` (:issue:`47578`)
 - Fixed regression in setting ``None`` or non-string value into a ``string``-dtype Series using a mask (:issue:`47628`)
+- Fixed regression in :func:`merge` throwing an error when passing a :class:`Series` with a multi-level name
```

Comment:
if there is no issue then the PR number should be added as the issue instead

(imo that should never happen though as an issue should always be raised to document the issue. there are many reasons to do this, help users find it, avoid duplicate issues, allow proper triage and give others an opportunity for discussion, and keep these out of the PR discussion as appears to have happen here)

Reply:
and also sometimes PRs go stale so the discussion in those cases is effectively lost.

Reply:
Good point, thanks for pointing out Simon!

Reply:
I agree with the sentiment. The only reason that I didn't create an issue in the first place was that I thought the bug happened because of a typo and that this PR should be straightforward. Aside from this, I want to point out two things:
1. Now that you've decided to reference the PR instead of the issue, you may want to add the base path of Github PRs to `sphinx.ext.extlinks`. From what I'm seeing, only the base path of issues is provided in the configuration file.
2. FYI, I ran into this bug when I was applying a complex function to a Series after a multi-level GroupBy.

Reply:
> 1. Now that you've decided to reference the PR instead of the issue, you may want to add the base path of Github PRs to `sphinx.ext.extlinks`. From what I'm seeing, only the base path of issues is provided in the configuration file.

GitHub itself handles those links fine. For example the link for this PR (https://github.com/pandas-dev/pandas/pull/47946), also works (redirects) if used with "issues": https://github.com/pandas-dev/pandas/issues/47946

Reply:
Uh I see. Thanks @jorisvandenbossche !